### PR TITLE
Firestore database and dashboard

### PR DESCRIPTION
**Pull request by ajay7231**
- [ ] **Dashboard** UI
- [ ] **Firestore** Database Implementation

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -45,3 +45,4 @@ testem.log
 .DS_Store
 Thumbs.db
 package-lock.json
+/assets/web
```
Comment (mridul-netizen) [inline .gitignore]: Try to leave an empty line at end of file wherever you find this symbol

File: src/app/app.component.css [hunk 2]
```diff
@@ -579,12 +579,12 @@ footer {
...
-   -webkit-box-shadow: 0 5px 15px rgba(0,0,0,0);
-    -moz-box-shadow: 0 5px 15px rgba(0,0,0,0);
-    -o-box-shadow: 0 5px 15px rgba(0,0,0,0);
-    box-shadow: 0 5px 15px rgba(0,0,0,0);
+.profile-modal .modal-content {
+  background: transparent;
+  -webkit-box-shadow: 0 5px 15px rgba(0, 0, 0, 0);
+  -moz-box-shadow: 0 5px 15px rgba(0, 0, 0, 0);
+  -o-box-shadow: 0 5px 15px rgba(0, 0, 0, 0);
+  box-shadow: 0 5px 15px rgba(0, 0, 0, 0);
```
Comment (mridul-netizen) [inline src/app/app.component.css:587]: Noice indentation

File: src/app/app.component.ts [hunk 3]
```diff
@@ -1,51 +1,49 @@
...
       animation: true,
       windowClass: 'form-modal',
     });
   }
-  openProfile(content) {
+
+  openProfile(content: any): void {
     this.profileModal = this.profileModalService.open(content, {
       animation: true,
       windowClass: 'profile-modal',
```
Comment (mridul-netizen) [inline src/app/app.component.ts:35]: We can move the `{ backdrop: static}` thing here if it is for the same ngbModal?

File: src/app/app.component.ts [hunk 4]
```diff
@@ -69,6 +67,5 @@ export class AppComponent implements OnInit {
   ForgotPassword(userName) {
     this.authService.ForgotPassword(userName);
     this.modal.close();
```
Comment (mridul-netizen) [inline src/app/app.component.ts]: Lets make a convention for Function names to be in camel case, try to define them as `forgotPassword` and not `ForgotPassword`

File: src/app/app.component.ts [hunk 5]
```diff
@@ -1,51 +1,49 @@
...
     config: NgbModalConfig,
     private modalService: NgbModal,
     private profileModalService: NgbModal,
+    public router: Router
   ) {
     config.backdrop = 'static';
   }
-  open(content) {
+  open(content: any): void {
     this.modal = this.modalService.open(content, {
```
Comment (mridul-netizen) [inline src/app/app.component.ts:25]: We can enable a rule here called strict checks.Actually using `any` as a type is a bad practice i.e why it is called Typescript ,we can get delay in getting response and some console errors with use of `any`.Should we enable strict checks at this point or after completion of the project ?WDYT?

File: src/app/auth/auth.guard.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+export class AuthGuard implements CanActivate {
+  constructor(public authService: AuthService, public router: Router) {}
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot
+  ): Observable<boolean> | Promise<boolean> | boolean {
+    if (this.authService.isLoggedIn() !== true) this.router.navigate(['home']);
+     return true;
+  }
```
Comment (mridul-netizen) [inline src/app/auth/auth.guard.ts:23]: This is so cool,Thanks!!

File: src/app/auth/auth.service.ts [hunk 7]
```diff
@@ -1,27 +1,29 @@
 import { Injectable, NgZone } from '@angular/core';
-import { User } from './models/user';
-// import  auth  from 'firebase/app';
+import { User } from './models/user.modal';
+import {patient} from './models/patient.model'
```
Comment (mridul-netizen) [inline src/app/auth/auth.service.ts:3]: `import { patient } from './models/patient.model'`

Comment (mridul-netizen) [inline src/app/auth/auth.service.ts]:
```suggestion
import { patient } from './models/patient.model'
```

File: src/app/auth/auth.service.ts [hunk 8]
```diff
@@ -125,9 +128,32 @@ export class AuthService {
...
+     const userRef: AngularFirestoreDocument<patient> = this.afs.doc(
+       `patients/${patient.id}`
+    );
+    // patient = {doctor_uid:user.uid, ...patient}
+    return userRef.set(patient, {
+      merge: true,
+    });
+    
+  }
+  UpdatePatient(patient:patient) {
```
Comment (mridul-netizen) [inline src/app/auth/auth.service.ts]: Try to use camel case for function names

File: src/index.html [hunk 9]
```diff
@@ -16,10 +16,11 @@
     href="https://cdnjs.cloudflare.com/ajax/libs/animate.css/4.1.1/animate.min.css"
   />
   
-
+  
 </head>
 <body class="mat-typography">
   <app-root></app-root>
   <script src="https://unpkg.com/ionicons@5.4.0/dist/ionicons.js"></script>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>
```
Comment (mridul-netizen) [inline src/index.html:24]: Do we still require jquery?

File: .gitignore (hunk 1 above)
Comment (ajay7231) [reply .gitignore]: Ok OMW

File: src/app/app.component.css (hunk 2 above)
Comment (ajay7231) [reply src/app/app.component.css:587]: Rip 😁

File: src/app/auth/auth.service.ts [hunk 10]
```diff
@@ -1,27 +1,29 @@
...
 @Injectable({
   providedIn: 'root',
 })
 export class AuthService {
   userData: any;
+  patients: Observable<any[]>;
 
   constructor(
     public afs: AngularFirestore,
     public afAuth: AngularFireAuth,
```
Comment (mridul-netizen) [inline src/app/auth/auth.service.ts:24]: `NgZone` is a cool thing ,but I'm not sure why is it required in this file,Please explain?

File: src/app/app.component.ts (hunk 3 above)
Comment (ajay7231) [reply src/app/app.component.ts:35]: yuss

**Pull request merged**